spec/requests/documents_spec.rb

```diff
@@ -5,7 +5,13 @@
 
 describe DocumentsController, type: :request do
 
-    let(:user) { create(:user) }
+    let!(:user) do
+        user = create(:user)
+        role = create(:role, :full_access)
+        user.roles << role
+        user.save
+        user
```

Comment:
```suggestion
        create(:user, role: create(:role, :full_access))
```
Would this work too?

Reply:
I tried the approach you suggested, but due to the intermediate relationship between user & role the factory can't assign a role directly:
```
users - user_roles - roles
```
I'm pretty sure there's a more elegan way of doing it within the factory itself tho